nemo/collections/nlp/modules/common/huggingface/huggingface_utils.py

```diff
@@ -31,71 +32,85 @@
+        'pretrained_model_list': BERT_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
     'distilbert': {
         'default': 'distilbert-base-uncased',
         'class': DistilBertEncoder,
         'config': DistilBertConfig,
-        'model_list': DISTILBERT_PRETRAINED_MODEL_ARCHIVE_LIST,
+        'pretrained_model_list': DISTILBERT_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
     'roberta': {
         'default': 'roberta-base',
         'class': RobertaEncoder,
         'config': RobertaConfig,
-        'model_list': ROBERTA_PRETRAINED_MODEL_ARCHIVE_LIST,
+        'pretrained_model_list': ROBERTA_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
     'albert': {
         'default': 'albert-base-v2',
         'class': AlbertEncoder,
         'config': AlbertConfig,
-        'model_list': ALBERT_PRETRAINED_MODEL_ARCHIVE_LIST,
+        'pretrained_model_list': ALBERT_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
 }
 
 
-def get_huggingface_lm_models_list() -> List[str]:
+def get_huggingface_lm_model(
+    model_type: str,
+    pretrained_model_name: Optional[str] = None,
+    config_dict: Optional[dict] = None,
+    config_file: Optional[str] = None,
+):
+    '''
+    Returns lm model instantiated with Huggingface
+    Args:
+        model_type: model type. Valid options are the keys in HUGGINGFACE_MODELS
+        pretrained_mode_name: specify this to instantiate pretrained model from Huggingface,
+            e.g. bert-base-cased. For entire list, see get_huggingface_pretrained_lm_models_list().
+        config_dict: model configuration dictionary used to instantiate Huggingface model from scratch
+        config_file: path to model configuration file used to instantiate Huggingface model from scratch
+    '''
+
+    # assumes that model configuration is correct, otherwise will pass on Huggingface's model instantiation error
+    model_class = HUGGINGFACE_MODELS[model_type]['class']
+    if config_file:
+        config_class = HUGGINGFACE_MODELS[model_type]['config']
+        return model_class(config_class.from_json_file(config_file))
+    elif config_dict:
+        config_class = HUGGINGFACE_MODELS[model_type]['config']
+        return model_class(config=config_class(**config_dict))
+    else:
+        return model_class.from_pretrained(pretrained_model_name)
+
+
+def get_huggingface_pretrained_lm_models_list() -> List[str]:
     '''
-    Returns the list of supported HuggingFace models
+    Returns the list of supported HuggingFace HUGGINGFACE_MODELS
     '''
     huggingface_models = []
-    for model in MODELS:
-        model_names = MODELS[model]['model_list']
+    for model in HUGGINGFACE_MODELS:
+        model_names = HUGGINGFACE_MODELS[model]['pretrained_model_list']
         huggingface_models.extend(model_names)
     return huggingface_models
+
+
+def get_all_huggingface_pretrained_lm_models_list() -> List[str]:
+    '''
+    Returns the list of all HuggingFace pretrained models
+    '''
+    return list(ALL_PRETRAINED_CONFIG_ARCHIVE_MAP.keys())
```

Comment:
so this will include models user will have to initialize directly by replacing get_lm_model()?

Reply:
we do not imply this function is connected to get_lm_model, only what the function says: all huggingface lm models.
So it includes all huggingface models user can instantiate using get_lm_model